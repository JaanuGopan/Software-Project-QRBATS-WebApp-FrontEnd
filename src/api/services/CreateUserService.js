import axios from "axios";
import ApiConstants from "../config/ApiConstants";

class CreateUserService {
  static async saveUser(
    firstName,
    lastName,
    email,
    password,
    userName,
    departmentId,
    role
  ) {
    const deparmentList = ["DEIE", "DCOM", "DMME", "DCEE", "DMENA"];
    const userRoleList = ["ADMIN", "LECTURER"];
    try {
      const response = await axios.post(ApiConstants.signupUrl, {
        firstName: firstName,
        lastName: lastName,
        email: email,
        password: password,
        userName: userName,
        departmentId: departmentId,
        role: role,
      });
      return response.data;
    } catch (error) {
      throw new Error("The User is not created. ", error);
    }
  }
}

export default CreateUserService;
