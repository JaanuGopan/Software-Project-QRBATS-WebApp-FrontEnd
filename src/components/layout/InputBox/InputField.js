import React from "react";
import "./inputfield.css";

const InputField = ({ placeholder, value, onChange, inputType }) => {
  return (
    <div className="input-with-icon">
      <input
        required
        type={inputType}
        className="input-field"
        placeholder={placeholder}
        value={value}
        onChange={onChange}
      />
    </div>
  );
};

export default InputField;
