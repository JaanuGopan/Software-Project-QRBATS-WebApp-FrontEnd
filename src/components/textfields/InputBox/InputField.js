import React from "react";
import "../../../pages/Event/EventCreation/EventCreation.css";

const InputField = ({ placeholder, value, onChange, inputType }) => {
  return (
    <div className="input-with-icon">
      <input
        required
        type={inputType}
        className="form-control mb-2"
        placeholder={placeholder}
        value={value}
        onChange={onChange}
      />
    </div>
  );
};

export default InputField;
