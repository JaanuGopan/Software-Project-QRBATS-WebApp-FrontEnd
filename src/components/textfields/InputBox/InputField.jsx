import React from 'react';
import '../../../pages/Event/EventCreation/EventCreation.css';

const InputField = ({ placeholder, value, onChange, inputType, error, helperText }) => {
  return (
    <div className="input-with-icon">
      <input
        required
        type={inputType}
        className={`form-control mb-2 ${error ? 'is-invalid' : ''}`}
        placeholder={placeholder}
        value={value}
        onChange={onChange}
      />
      {error && <div className="invalid-feedback">{helperText}</div>}
    </div>
  );
};

export default InputField;
