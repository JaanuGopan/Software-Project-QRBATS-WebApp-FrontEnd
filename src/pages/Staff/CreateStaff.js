import React, { useState } from "react";
import "./StaffA.css";
import axios from "axios";
import Designer from "../../assets/Images/Designer.jpeg";
import InputField from "../../components/textfields/InputBox/InputField";
import { IoMdCloseCircleOutline } from "react-icons/io";
import CreateUserService from "../../api/services/CreateUserService";

const CreateStaff = ({ handlecloseCreateStaffWindow }) => {
  const [firstname, setFirstName] = useState("");
  const [lastname, setLastName] = useState("");
  const [email, setEmail] = useState("");
  const [password, setPassword] = useState("");
  const [confirmPassword, setConfirmPassword] = useState("");
  const [userName, setUserName] = useState("");
  const [userRole, setUserRole] = useState("");
  const [departmentId, setDepartmentId] = useState("");

  const deparmentList = ["DEIE", "DCOM", "DMME", "DCEE", "DMENA"];
  const userRoleList = ["ADMIN", "LECTURER"];

  const handleSubmit = async (e) => {
    e.preventDefault();
    try {
      const response = CreateUserService.saveUser(
        firstname,
        lastname,
        email,
        password,
        userName,
        deparmentList.indexOf(departmentId) + 1,
        userRoleList.indexOf(userRole)
      );

      console.error("Login success...");
      handlecloseCreateStaffWindow();
    } catch (error) {
      console.error("Login failed", error);
    }
  };

  return (
    <div className="staff-signup-main-container">
      <div
        className="closeCreateEventWindow"
        onClick={handlecloseCreateStaffWindow}
      >
        <IoMdCloseCircleOutline />
      </div>
      <p className="staff-head1">Create Staff</p>
      <div className="staff-login-container">
        <div className="staff-image-container">
          <img src={Designer} className="staff-logo" alt="Logo" />
        </div>
        <div className="form-container">
          <form onSubmit={handleSubmit}>
            <div className="choice-input mb-3">
              <select
                value={userRole}
                onChange={(e) => setUserRole(e.target.value)}
                className="student-select-input"
              >
                <option value="">Select user role</option>
                {userRoleList.map((option, index) => (
                  <option key={index} value={option}>
                    {option}
                  </option>
                ))}
              </select>
            </div>
            {/* First Name Input */}
            <InputField
              placeholder="Enter your first name"
              value={firstname}
              onChange={(e) => setFirstName(e.target.value)}
              inputType="text"
            />

            {/* Last Name Input */}
            <InputField
              placeholder="Enter your last name"
              value={lastname}
              onChange={(e) => setLastName(e.target.value)}
              inputType="text"
            />

            {/* Email Input */}
            <InputField
              placeholder="Enter your email"
              value={email}
              onChange={(e) => setEmail(e.target.value)}
              inputType="text"
            />

            <InputField
              placeholder="Enter your user name"
              value={userName}
              onChange={(e) => setUserName(e.target.value)}
              inputType="text"
            />

            {/* Password Input */}
            <InputField
              placeholder="Enter your password"
              value={password}
              onChange={(e) => setPassword(e.target.value)}
              inputType="password"
            />
            <InputField
              placeholder="Confirm password"
              value={confirmPassword}
              onChange={(e) => setConfirmPassword(e.target.value)}
              inputType="password"
            />

            <div className="choice-input mb-3">
              <select
                value={departmentId}
                onChange={(e) => setDepartmentId(e.target.value)}
                className="student-select-input"
              >
                <option value="">Select the department</option>
                {deparmentList.map((option, index) => (
                  <option key={index} value={option}>
                    {option}
                  </option>
                ))}
              </select>
            </div>
            <button type="submit" className="btn btn-primary w-100">
              Add Staff
            </button>
          </form>
        </div>
      </div>
    </div>
  );
};

export default CreateStaff;
