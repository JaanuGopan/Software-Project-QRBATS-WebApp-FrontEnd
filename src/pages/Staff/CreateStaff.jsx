import React, { useState } from 'react';
import './StaffA.css';
import InputField from '../../components/textfields/InputBox/InputField';
import { IoMdCloseCircleOutline } from 'react-icons/io';
import CreateUserService from '../../api/services/CreateUserService';
import Select from 'react-select';
import { toast } from 'react-toastify';
import { CircularProgress } from '@mui/material';
import Department from '../../utils/Department';

const CreateStaff = ({ handleCloseCreateStaffWindow, reloadStaffList }) => {
  const [firstName, setFirstName] = useState('');
  const [lastName, setLastName] = useState('');
  const [email, setEmail] = useState('');
  const [password, setPassword] = useState('');
  const [confirmPassword, setConfirmPassword] = useState('');
  const [userName, setUserName] = useState('');
  const [userRole, setUserRole] = useState('');
  const [departmentId, setDepartmentId] = useState('');

  const departmentList = Department.departmentList;
  const userRoleList = ['ADMIN', 'LECTURER'];

  const notifySuccess = () => toast.success('Successfully Staff Created!');

  const [loadingCreateStaff, setLoadingCreateStaff] = useState(false);

  const handleInputValidation = () => {
    const emailPattern = /^[^\s@]+@[^\s@]+\.[^\s@]+$/;

    if (firstName.trim() === '') {
      toast.error('Please enter first name');
      return false;
    }
    if (lastName.trim() === '') {
      toast.error('Please enter last name');
      return false;
    }
    if (email.trim() === '' || !emailPattern.test(email)) {
      toast.error('Please enter a valid email');
      return false;
    }
    if (userName.trim() === '') {
      toast.error('Please enter username');
      return false;
    }
    if (password.trim() === '') {
      toast.error('Please enter password');
      return false;
    }
    if (confirmPassword.trim() === '') {
      toast.error('Please confirm password');
      return false;
    }
    if (password !== confirmPassword) {
      toast.error('Passwords do not match');
      return false;
    }
    if (!departmentId) {
      toast.error('Please select department');
      return false;
    }
    if (!userRole) {
      toast.error('Please select user role');
      return false;
    }
    return true;
  };

  const handleSubmit = async (e) => {
    e.preventDefault();
    if (!handleInputValidation()) {
      return;
    }
    try {
      setLoadingCreateStaff(true);
      const response = await CreateUserService.saveUser(
        firstName,
        lastName,
        email,
        password,
        userName,
        departmentList.indexOf(departmentId.value) + 1,
        userRoleList.indexOf(userRole.value)
      );
      if (response.status === 200) {
        notifySuccess();
        reloadStaffList();
        handleCloseCreateStaffWindow();
        toast.success('User created successfully!');
      } else if (response.status === 400) {
        toast.error(response.data);
      }
    } catch (error) {
      console.error('User creation failed', error);
      toast.error('Error in user creation.');
    } finally {
      setLoadingCreateStaff(false);
    }
  };

  return (
    <div className="staff-signup-main-container">
      <div className="staff-update-title-close-button">
        <h3 className="staff-update-title">Create User</h3>
        <div className="staff-update-close-button" onClick={handleCloseCreateStaffWindow}>
          <IoMdCloseCircleOutline id="close-icon" />
        </div>
      </div>

      <div className="staff-login-container">
        <div className="form-container">
          <form onSubmit={handleSubmit}>
            <div className="staff-creation-input">
              <label>User Role</label>
              <div className="staff-creation-input-field">
                <Select
                  required
                  value={userRole}
                  onChange={(e) => setUserRole(e)}
                  options={userRoleList.map((role) => ({
                    value: role,
                    label: role,
                  }))}
                  placeholder={'Select User Role'}
                />
              </div>
            </div>
            <div className="staff-creation-input">
              <label>First Name</label>
              <div className="staff-creation-input-field">
                <InputField
                  placeholder="Enter your first name"
                  value={firstName}
                  onChange={(e) => setFirstName(e.target.value)}
                  inputType="text"
                />
              </div>
            </div>
            <div className="staff-creation-input">
              <label>Last Name</label>
              <div className="staff-creation-input-field">
                <InputField
                  placeholder="Enter your last name"
                  value={lastName}
                  onChange={(e) => setLastName(e.target.value)}
                  inputType="text"
                />
              </div>
            </div>

            <div className="staff-creation-input">
              <label>Email</label>
              <div className="staff-creation-input-field">
                <InputField
                  placeholder="Enter your email"
                  value={email}
                  onChange={(e) => setEmail(e.target.value)}
                  inputType="text"
                />
              </div>
            </div>
            <div className="staff-creation-input">
              <label>User Name</label>
              <div className="staff-creation-input-field">
                <InputField
                  placeholder="Enter your user name"
                  value={userName}
                  onChange={(e) => setUserName(e.target.value)}
                  inputType="text"
                />
              </div>
            </div>

            <div className="staff-creation-input">
              <label>Password</label>
              <div className="staff-creation-input-field">
                <InputField
                  placeholder="Enter your password"
                  value={password}
                  onChange={(e) => setPassword(e.target.value)}
                  inputType="password"
                />
              </div>
            </div>
            <div className="staff-creation-input">
              <label>Confirm Password</label>
              <div className="staff-creation-input-field">
                <InputField
                  placeholder="Confirm password"
                  value={confirmPassword}
                  onChange={(e) => setConfirmPassword(e.target.value)}
                  inputType="password"
                />
              </div>
            </div>
            <div className="staff-creation-input">
              <label>Department</label>
              <div className="staff-creation-input-field">
                <Select
                  required
                  value={departmentId}
                  onChange={(e) => setDepartmentId(e)}
                  options={departmentList.map((dept) => ({
                    value: dept,
                    label: dept,
                  }))}
                  placeholder={'Select Department'}
                />
              </div>
            </div>
            <div className="create-staff-create-button">
              {loadingCreateStaff ? (
                <div className="d-flex justify-content-center align-items-center">
                  <CircularProgress />
                </div>
              ) : (
                <button type="submit" className="btn btn-success">
                  Create Staff
                </button>
              )}
            </div>
          </form>
        </div>
      </div>
    </div>
  );
};

export default CreateStaff;
