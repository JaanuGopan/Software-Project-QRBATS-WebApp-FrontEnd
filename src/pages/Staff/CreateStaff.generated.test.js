import renderer from 'react-test-renderer';
import React, { useState } from "react";

import axios from "axios";
import Designer from "../../assets/Images/Designer.jpeg";
import InputField from "../../components/textfields/InputBox/InputField";
import { IoMdCloseCircleOutline } from "react-icons/io";
import CreateUserService from "../../api/services/CreateUserService";
import CreateStaff from "./CreateStaff";

jest.mock("./StaffA.css");
jest.mock("axios");
jest.mock("../../assets/Images/Designer.jpeg");
jest.mock("../../components/textfields/InputBox/InputField");
jest.mock("react-icons/io");
jest.mock("../../api/services/CreateUserService");

const renderTree = tree => renderer.create(tree);
describe('<CreateStaff>', () => {
  it('should render component', () => {
    expect(renderTree(<CreateStaff 
    />).toJSON()).toMatchSnapshot();
  });
  
});