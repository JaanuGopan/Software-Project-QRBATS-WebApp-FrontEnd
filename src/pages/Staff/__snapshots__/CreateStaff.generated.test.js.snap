// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CreateStaff> should render component 1`] = `
<div
  className="staff-signup-main-container"
>
  <div
    className="closeCreateEventWindow"
  />
  <p
    className="staff-head1"
  >
    Create Staff
  </p>
  <div
    className="staff-login-container"
  >
    <div
      className="staff-image-container"
    >
      <img
        alt="Logo"
        className="staff-logo"
        src="Designer.jpeg"
      />
    </div>
    <div
      className="form-container"
    >
      <form
        onSubmit={[Function]}
      >
        <div
          className="choice-input mb-3"
        >
          <select
            className="student-select-input"
            onChange={[Function]}
            value=""
          >
            <option
              value=""
            >
              Select user role
            </option>
            <option
              value="ADMIN"
            >
              ADMIN
            </option>
            <option
              value="LECTURER"
            >
              LECTURER
            </option>
          </select>
        </div>
        <div
          className="choice-input mb-3"
        >
          <select
            className="student-select-input"
            onChange={[Function]}
            value=""
          >
            <option
              value=""
            >
              Select the department
            </option>
            <option
              value="DEIE"
            >
              DEIE
            </option>
            <option
              value="DCOM"
            >
              DCOM
            </option>
            <option
              value="DMME"
            >
              DMME
            </option>
            <option
              value="DCEE"
            >
              DCEE
            </option>
            <option
              value="DMENA"
            >
              DMENA
            </option>
          </select>
        </div>
        <button
          className="btn btn-primary w-100"
          type="submit"
        >
          Add Staff
        </button>
      </form>
    </div>
  </div>
</div>
`;
